### PR TITLE
[CID-1204] https://projects.engineering.redhat.com/browse/CID-1204.

### PR DESCRIPTION
**Pull request by ggallen**

### CODE REVIEW

File: src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java [hunk 1]
```diff
@@ -136,54 +134,43 @@ public StepExecution start(StepContext context) throws Exception {
...
-                }
+}
             });
             return false;
         }
 
         @Override
         public void stop(@Nonnull Throwable cause) throws Exception {
             if (task != null) {
-                getContext().get(TaskListener.class).getLogger().println("in stop of watcher");
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java]: why is this being removed?

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java [hunk 2]
```diff
@@ -151,17 +146,17 @@ public String waitforCIMessage(Run<?, ?> build, Launcher launcher, TaskListener
             return null;
         }
 
-        JMSMessagingWorker worker =
-                provider.createWorker(overrides, build.getParent().getName());
+        JMSMessagingWorker worker = provider.createWorker(overrides, build.getParent().getName());
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java]: only formatting change here?

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java [hunk 3]
```diff
@@ -89,30 +115,13 @@ public String getMessageContent() {
...
         this.messageContent = messageContent;
     }
-    public boolean isFailOnError() {
+    public boolean getFailOnError() {
         return failOnError;
     }
-
-    @DataBoundSetter
-    public void setFailOnError(boolean failOnError) {
+    public void setFailOnErrro(boolean failOnError) {
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java]: typo

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java [hunk 4]
```diff
@@ -89,30 +115,13 @@ public String getMessageContent() {
     public void setMessageContent(String messageContent) {
         this.messageContent = messageContent;
     }
-    public boolean isFailOnError() {
+    public boolean getFailOnError() {
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java]: why the change from is() to get()? usually a boolean is an is()

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java [hunk 5]
```diff
@@ -81,18 +86,9 @@ public CIMessageSubscriberBuilder(String providerName,
...
-
-    public CIMessageSubscriberBuilder(String providerName, MessagingProviderOverrides overrides,
-                                      String selector, List<MsgCheck> checks, Integer timeout) {
-        this.providerName = providerName;
-        this.overrides = overrides;
-        this.selector = selector;
-        if (checks == null) {
-            checks = new ArrayList<>();
+        if (timeout == null) {
+            timeout = DEFAULT_TIMEOUT_IN_MINUTES;
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java]: good catch

File: src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java (hunk 1 above)
Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java]: To me it appears to be a debug message that got left in accidentally.  I can put it back.

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java (hunk 2 above)
Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java]: Yes.

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java (hunk 4 above)
Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java]: I will change this back.

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java (hunk 3 above)
Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIMessageNotifier.java]: Good catch. I will fix this.

**Comment by ggallen** [comment]
These changes are mainly to allow changes in the RH CI plugin to expose steps to the pipeline.  To keep from duplicating code, I refactored some of the non-pipeline code so it can be re-used by pipeline steps.

File: src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java (hunk 2 above)
Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIMessageSubscriberBuilder.java]: I'll undo this too.

**Comment by ggallen** [comment]
Closing this review and will submit another PR.
